packages/core/utils/src/dml/types.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * The supported data types
+ */
+export type KnownDataTypes =
+  | "string"
+  | "boolean"
+  | "enum"
+  | "number"
+  | "date"
+  | "dateTime"
```

Comment:
thought: Should we eliminate `dateTime` and enforce that `date` applies the `timestamptz` type?

Reply:
What if I need a `date` data-type without the time?

Reply:
We haven't had the need for it yet in any of our modules, which motivated the suggestion. I personally think the more we can reduce the amount of options the better. Ofc. it should not come at the cost of being too inflexible, but I don't think that's the case here.

Would be curious to hear what others think.

Reply:
We can always add the other type further down the line, if we see a demand for it.

Reply:
If we want to add it back later, then we should go with `dateTime` and add back `date` later. Because, otherwise it will be a breaking change.

Reply:
Imo. we don't need a direct translation between the schema types and what gets applied under the hood. I would use the most intuitive naming for the most common use cases.

So if we need to add the `date` data type later, we could call it something else, e.g. `simpleDate` or `date(type: "simple")`.

Reply:
I won't push much. But is there any reason that `date` feels better than `dateTime`. Also, introducing a very confusing naming like `simpleDate` won't help much either.

Reply:
I believe dateTime would be more appropriate, users are also more use to see dataTime when it comes to date + time IMO, simple date I have never seen date anywhere which would add confusion if we add that as it does not exists either in typescript or the DB types or other languages. wdyt?

Reply:
> But is there any reason that date feels better than dateTime

It's not a strong opinion, but I just find it nicer. I am also fine with `dateTime`. It is a very minor difference, I agree. I think it's similar to using `json` over `jsonb`.

@srindom, do you have any thoughts on this?

Reply:
ahah 😄 no problem I see what you mean, but we still will need to find a name which will be completely different from what users are use too IMO

Reply:
I would also go with dateTime, as it is more explicit and obvious what it holds (you don't need to wonder what precision this is for).  Another option is to have `date`, which takes in modifiers such as `time | date | full` or something along those lines.

Reply:
QQ: When you guys say it is more obvious and known to developers, what usage are you referring to?

Reply:
@olivermrbl In general:

Anytime I want to represent `time`, `date` and `datetime` separately. I will use the following naming.

- `time` - To represent time without date
- `date` - To represent date without time
- `dateTime` - To represent date time as a timestamp (precision option can be configured on this )

Yes, usually these terms are connected to a database. But even outside that if I have to talk about these three unit of values, I will use the same naming and won't invent it.

Reply:
On my side I am referring to many languages, database, packages an browser as well

Reply:
I don't have strong opinions here - fine with having both date and datetime. 

Would maybe suggest we are opinionated about including timezone on datetime though?

Reply:
I believe timezone is always good to have to manage date properly so fine with including it